processor/filterprocessor/filter_processor.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filterprocessor
+
+import (
+	"context"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+	"go.opentelemetry.io/collector/internal/processor/filtermetric"
+)
+
+type filterMetricProcessor struct {
+	cfg          *Config
+	capabilities component.ProcessorCapabilities
+	next         consumer.MetricsConsumerOld
+	include      *filtermetric.Matcher
+	exclude      *filtermetric.Matcher
+}
+
+var _ component.MetricsProcessorOld = (*filterMetricProcessor)(nil)
+
+func newFilterMetricProcessor(next consumer.MetricsConsumerOld, cfg *Config) (*filterMetricProcessor, error) {
+	inc, err := createMatcher(cfg.Metrics.Include)
+	if err != nil {
+		return nil, err
+	}
+
+	exc, err := createMatcher(cfg.Metrics.Exclude)
+	if err != nil {
+		return nil, err
+	}
+
+	return &filterMetricProcessor{
+		cfg:          cfg,
+		capabilities: component.ProcessorCapabilities{MutatesConsumedData: false},
```

Comment:
Could someone who is familiar with shared ownership comment on if this is correct? The filter processor is not mutating the underlying data of any of the *metricspb.Metric that get passed through, but is changing the collection of Metrics that gets passed to the next processor.

Ideally, the filterprocessor would not need to copy incoming data because it is not modifying the metrics itself, just which ones continue along a particular processor pipeline.

Reply:
The way how it is implemented (you don't change the incoming slice but construct a different one, returning false is ok).

Reply:
Ah, good point. I forgot that it was already making a new slice.